src/Microsoft.AspNet.WebHooks.Receivers/Extensions/WebHookHandlerContextExtensions.cs

```diff
@@ -30,8 +30,10 @@ public static T GetDataOrDefault<T>(this WebHookHandlerContext context)
                 return default(T);
             }
 
-            if (context.Data is JToken && !typeof(JToken).IsAssignableFrom(typeof(T)))
+            if (context.Data is JToken && !typeof(T).IsAssignableFrom(context.Data.GetType()))
```

Comment:
Add tests to confirm potential failure cases e.g. `T` is `JArray` but the current object is `JObject`.

Reply:
Testing showed this change only affects corner cases. It's less efficient in some of those cases e.g. where `Data` is a `JObject` and `T` is `JArray`. Will attempt the impossible conversion and catch the thrown `Exception` rather than simply returning `null`. Don't see a compelling reason to do more than add comments and tests.